src/psyclone/gocean1p0.py

```diff
@@ -251,9 +259,12 @@ def gen_code(self, parent):
                 invoke.schedule.children.insert(1, assign1)
                 invoke.schedule.children.insert(2, assign2)
 
-            invoke.schedule.root.lower_to_language_level()
-            for child in invoke.schedule.root.children:
-                parent.add(PSyIRGen(parent, child))
+        # Lower the GOcean PSyIR to language level so it can be visited
+        # by the backends
+        invoke.schedule.root.lower_to_language_level()
```

Comment:
`invoke.schedule.root` probably gives us a `FileContainer` node now which will then only have a `Container` as a child? Therefore, is the following loop over children needed or have I missed something?

Reply:
The PSy-layer is manually created and doesn't have a FileContainer in the top at this point, just the module container. The loop over children is needed because a module-inlinded CodedKerns will add a new routine into the top container. I added a comment.